File: LeetCode/LeetCode Practice Problems/1381- Design a Stack With Increment Operation.cpp
class CustomStack {
public:
    int mx = 0;
    stack<int> st;
    CustomStack(int maxSize) {
        mx = maxSize;
    }
    
    void push(int x) {
        if(st.size() < mx)
            st.push(x);
    }
    
    int pop() {
        if(st.empty())
            return -1;
        
        int val = st.top();
        st.pop();
        
        return val;
    }
    
    void increment(int k, int val) {
        stack<int> s;
        
        while(!st.empty()) {
            int v = st.top();
            st.pop();
            s.push(v);
        }
        
        while(!s.empty()) {
            int v = s.top();
            if(k > 0)
                v+=val,k--;
            st.push(v);
            s.pop();
        }
        
    }
};

File: LeetCode/LeetCode Practice Problems/1381-Design a Stack With Increment Operation.java
class CustomStack {
    int a[],sz,cur;
    
    public CustomStack(int maxSize) {
        a = new int[maxSize];
        cur = -1;
        sz = maxSize;
    }
    
    public void push(int x) {
        if(cur+1 != sz) {
            a[++cur] = x;
        }
    }
    
    public int pop() {
        if(cur == -1) {
            return -1;
        }
        int val = a[cur--];
        return val;
    }
    
    public void increment(int k, int val) {
        for(int i=0;i<=Math.min(k-1,cur);i++) {
            a[i] += val;
        }
    }
}
